nb-api/src/main/java/io/nosqlbench/components/NBComponentSubScope.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.components;
+
+public class NBComponentSubScope implements AutoCloseable {
```

Comment:
Not sure I understand the purpose of this class. I see it adding in AutoCloseable but why not simply have that implemented by NBBaseComponent?

Reply:
We did implement NBBaseComponent as such, but it seems that this causes some pain for developers with verbose warnings from static analysis anywhere an AutoCloseable is used outside of the try with resources. We'll probably move it to a similar form outside of the official AutoCloseable type.